base_classes/nyaml/NXbeam_optical_time_resolved.yaml

```diff
@@ -0,0 +1,50 @@
+doc: |
+  Properties of an optical time resolved beam at a given location.
+category: base
+(NXbeam_optical)NXbeam_optical_time_resolved:
+  pulse_energy(NX_FLOAT):
+    unit: NX_ENERGY
+    doc: |
+      Energy of a single pulse at the diagnostic point
+  average_power(NX_FLOAT):
+    unit: NX_POWER
+    doc: |
+      Average power at the diagnostic point
+  fluence(NX_FLOAT):
+    unit: NX_ANY
+    doc: |
+      Incident fluence at the diagnostic point
+    \@units:
+      doc: |
+        Here: SI units are 'J/m2', customary 'mJ/cm2'.
+  pulse_duration(NX_FLOAT):
+    unit: NX_TIME
+    doc: |
+      FWHM duration of the pulses at the diagnostic point
+  frog_trace(NX_FLOAT):
+    doc: |
+      FROG trace of the pulse.
+    dimensions:
+      rank: 2
+      dim: [[1, nx], [2, ny]]
+  frog_delays(NX_FLOAT):
+    unit: NX_TIME
+    doc: |
+      Horizontal axis of a FROG trace, i.e. delay.
+    dimensions:
+      rank: 1
+      dim: [[1, nx]]
+  frog_frequencies(NX_FLOAT):
+    unit: NX_FREQUENCY
+    doc: |
+      Vertical axis of a FROG trace, i.e. frequency.
+    dimensions:
+      rank: 1
+      dim: [[1, ny]]
+  chirp_type(NX_CHAR):
+    doc: |
+      The type of chirp implemented
+  chirp_GDD(NX_FLOAT):
+    unit: NX_TIME
+    doc: |
+      Group delay dispersion of the pulse for linear chirp
```

Comment:
If there is a linear chirp there is likely also a nonlinear chirp, would a base class NXsignal_processing_descriptor_chirp make sense (dont take that naive name of this base class for granted) but read "between the lines" conceptually?

Same story for frog?

I see one problem/inconvenience of a finer granularization maybe people need to navigate deeper but the benefit is stronger modularization, I am still not convinced though that I am overdoing it here with the balance between convenience and granularization